File: Algorithms/057_InsertInterval.py
# Definition for an interval.
# class Interval(object):
#     def __init__(self, s=0, e=0):
#         self.start = s
#         self.end = e

class Solution(object):
    def insert(self, intervals, newInterval):
        """
        :type intervals: List[Interval]
        :type newInterval: Interval
        :rtype: List[Interval]
        """
        result = []
        i = 0
        while i < len(intervals):
            if newInterval.start > intervals[i].end:
                result.append(intervals[i])
            elif newInterval.end < intervals[i].start:
                break
            else:
                if newInterval.end < intervals[i].end:
                    newInterval.end = intervals[i].end
                if newInterval.start > intervals[i].start:
                    newInterval.start = intervals[i].start
            i += 1
        result.append(newInterval);
        result.extend(intervals[i:])
        return result

File: Algorithms/java/057.InsertInterval.java
public class Solution {
    public List<Interval> insert(List<Interval> intervals, Interval newInterval) {
        List<Interval> result = new ArrayList<>();
        int i = 0;
        while (i < intervals.size()) {
            if (newInterval.start > intervals.get(i).end) {
                result.add(intervals.get(i));
            } else if (newInterval.end < intervals.get(i).start) {
                break;
            } else {
                if (newInterval.end < intervals.get(i).end) {
                    newInterval.end = intervals.get(i).end;
                }
                if (newInterval.start > intervals.get(i).start) {
                    newInterval.start = intervals.get(i).start;
                }
            }
            i ++;
        }
        result.add(newInterval);
        while (i < intervals.size()) {
            result.add(intervals.get(i));
            i ++;
        }
        return result;
    }
}
